src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -100,25 +125,76 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 while (userName.Length == 0);
             }
 
-            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
-            );
+            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName);
 
-            //
             // now, prompt for the password
-            //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+            do
             {
-                return null;
+                WriteToConsole(passwordPrompt, true);
+                password = ReadLineAsSecureString();
+                if (password == null)
+                {
+                    return null;
+                }
             }
+            while (password.Length == 0);
 
-            WriteLineToConsole();
+            if (reenterPassword)
+            {
+                reenterPasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ReenterPassword, userName);
+                passwordMismatch = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_PasswordMismatch);
+                do
+                {
+                    // now, prompt to re-enter the password.
+                    WriteToConsole(reenterPasswordPrompt, true);
+                    confirmPassword = ReadLineAsSecureString();
+                    if (confirmPassword == null)
+                    {
+                        return null;
+                    }
 
-            cred = new PSCredential(userName, password);
+                    if(SecureStringEquals(password, confirmPassword))
```

Comment:
```suggestion
                    if (SecureStringEquals(password, confirmPassword))
```

Reply:
Sure. I will fix it. Anything else that I need to work on ?